uddl_test/uddl_test_formatting/universal-data-definition-language-formatter.test.ts

```diff
@@ -46,14 +43,84 @@ describe("Universal Data Definition Language Formatter", () => {
         
         };
         
+    }
+    
+}`,
+    });
+  });
+
+  it("should format cdm with conceptual entity", async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before:
+        'dm PPT "Base data structures to support People, Places and Things"{cdm Conceptual "Need to start at Conceptual Level"{centity AddressableEntity "Any entity that is addressable in some way" {};}}',
+      after: `dm PPT "Base data structures to support People, Places and Things"
+{
+
+    cdm Conceptual "Need to start at Conceptual Level"
+    {
+    
         centity AddressableEntity "Any entity that is addressable in some way"
         {
         
         };
         
     }
     
-}`
+}`,
     });
   });
-});
\ No newline at end of file
+
+  it('should format cdm with conceptual basis',async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before: 
+      'dm PPT "Base data structures to support People, Places and Things"{cdm Conceptual "Need to start at Conceptual Level"{basis uddlBasis "Formating conceptual basis entity"; }}',
+      after: `dm PPT "Base data structures to support People, Places and Things"
+{
+
+    cdm Conceptual "Need to start at Conceptual Level"
+    {
+    
+        basis uddlBasis "Formating conceptual basis entity";
+    }
+    
+}`,
+    })    
+  })
+
+  it('should format cdm with conceptual domain',async () => {
+    await universalDataDefinitionLanguageFormatting({
+      before: 
+      'dm PPT "Base data structures to support People, Places and Things"{cdm Conceptual "Need to start at Conceptual Level"{domain NaturalPerson "Base definition of a natural person"; }}',
+      after: `dm PPT "Base data structures to support People, Places and Things"
+{
+
+    cdm Conceptual "Need to start at Conceptual Level"
+    {
+    
```

Comment:
Do you really want all these blank lines after formatting? You've got a blank line after every '{' and '}'.  That's going to make some files a lot longer - which will force users to scroll. When people have to scroll, they forget what they can't see - which makes it harder to understand complex content.

Reply:
The test would not pass without the blank lines in the after block

Reply:
When a test does not pass, you have a choice: change the test (what you did) or change the code being tested.  The point I'm making above is that the formatter currently inserts those blank lines - which is why your test needs the blank lines to pass. My question is: do you want the formatter to do that?  The goal of formatting is to put the text into a format that is easy to read and understand.  Every decision you make about how the formatter works should be made with that in mind.